### PR TITLE
Check divu

### PR DESCRIPTION
**Pull request by nickwimer**
This branch implements the pressure rise due to enclosed mass injection by summing up velocity flux on all boundary faces and incorporates the correction in adjustPandDivU.

### CODE REVIEW

File: Source/PeleLMAdvection.cpp [hunk 1]
```diff
@@ -631,6 +631,15 @@ void PeleLM::computeScalarAdvTerms(std::unique_ptr<AdvanceAdvData> &advData)
    if (m_do_massBalance && (m_sdcIter == m_nSDCmax)) {
       addMassFluxes(GetArrOfConstPtrs(fluxes[0]),geom[0]);
    }
+   // Compute face domain integrals for RhoH and RhoY
+   if (m_sdcIter == m_nSDCmax) {
+      addRhoHFluxes(GetArrOfConstPtrs(fluxes[0]),geom[0]);
+      addRhoYFluxes(GetArrOfConstPtrs(fluxes[0]),geom[0]);
+   }
+   // Compute face domain integral for U
+   if (m_sdcIter == m_nSDCmax) {
```
Comment (esclapez) [inline Source/PeleLMAdvection.cpp]: Shoudn't the Umac divergence be done at every SDC step?

File: Source/PeleLMEos.cpp [hunk 2]
```diff
@@ -290,6 +303,15 @@ PeleLM::adjustPandDivU(std::unique_ptr<AdvanceAdvData> &advData)
...
 
+
+    Vector<MultiFab> dummy(finest_level+1);
+    for (int lev = 0; lev <= finest_level; ++lev) {
+       dummy[lev].define(grids[lev], dmap[lev], 1, 0, MFInfo(), *m_factory[lev]);
+       dummy[lev].setVal(1.0);
+    }
+
+    m_uncoveredVol = MFSum(GetVecOfConstPtrs(dummy),0);
+
```
Comment (esclapez) [inline Source/PeleLMEos.cpp]: m_uncoveredVol should already be computed when you come here no ?

File: Source/PeleLMEos.cpp [hunk 3]
```diff
@@ -298,19 +320,29 @@ PeleLM::adjustPandDivU(std::unique_ptr<AdvanceAdvData> &advData)
...
+    m_pNew = m_pOld + m_dt * (Sbar - divu_vol)/Thetabar;
+    m_dp0dt = (Sbar - divu_vol)/Thetabar;
+
 
     // subtract \tilde{theta} * Sbar / Thetabar from divu
     for (int lev = 0; lev <= finest_level; ++lev) {
 #ifdef AMREX_USE_OMP
 #pragma omp parallel if (Gpu::notInLaunchRegion())
 #endif
+
```
Comment (esclapez) [inline Source/PeleLMEos.cpp]: This empty line might mess with #pragma omp parallel if

File: Source/PeleLMTemporals.cpp [hunk 4]
```diff
@@ -2,19 +2,37 @@
...
+      m_RhoYOld[n] = MFSum(GetVecOfConstPtrs(getSpeciesVect(a_time)),n);
+      for (int idim = 0; idim < AMREX_SPACEDIM; idim++) {
+         m_domainRhoYFlux[2*n*AMREX_SPACEDIM+2*idim] = 0.0;
+         m_domainRhoYFlux[1+2*n*AMREX_SPACEDIM+2*idim] = 0.0;
+      }
+   }
+   for (int idim = 0; idim < AMREX_SPACEDIM; idim++) {
+      m_domainUmacFlux[2*idim] = 0.0;
+      m_domainUmacFlux[2*idim+1] = 0.0;
+   }
```
Comment (esclapez) [inline Source/PeleLMTemporals.cpp]: This should be done regardless of the `m_do_temporal` status right ?

File: Source/PeleLMAdvection.cpp (hunk 1 above)
Comment (nickwimer) [reply Source/PeleLMAdvection.cpp]: Yeah, good point. Changing...

File: Source/PeleLMEos.cpp (hunk 3 above)
Comment (nickwimer) [reply Source/PeleLMEos.cpp]: deleting...

File: Source/PeleLMTemporals.cpp (hunk 4 above)
Comment (nickwimer) [reply Source/PeleLMTemporals.cpp]: pulling out and placing in PeleLMAdvance

**Pull request merged**